src/service/repository/users.rs

```diff
@@ -60,27 +60,68 @@ impl User {
 
     pub fn get_user(
         connection: &mut PgConnection,
+        user_id: Option<&Uuid>,
         username: Option<&str>,
         email: Option<&str>,
         github_username: Option<&str>,
     ) -> Result<User> {
         use crate::schema::users::dsl::{
             email as email_col, github_username as github_username_col, username as username_col,
         };
-        let user = users::table
-            .filter(username_col.eq(username.unwrap_or_default()))
-            .or_filter(email_col.eq(email.unwrap_or_default()))
-            .or_filter(github_username_col.eq(github_username.unwrap_or_default()))
-            .first::<User>(connection)
-            .optional()
-            .map_err(|e| {
-                error!("Error getting user: {}", e);
-                FailedToGetUser(GetUserError(e))
-            })?;
 
-        user.ok_or_else(|| {
-            error!("User not found");
-            UserNotFound.into()
-        })
+        match (user_id, username, email, github_username) {
```

Comment:
The current `get_user` implementation has two issues:

1. It doesn't correctly handle cases where 2-3 parameters are provided.
2. The error message is misleading when multiple parameters are given.

Reply:
addressed https://github.com/Extheoisah/hxckr-core/pull/19/commits/a110e98ad5bdbd5b945d61bb8b9206a37b54eb98